cmd/snap-confine/mount-support.c

```diff
@@ -301,7 +305,31 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		}
 		sc_must_snprintf(dst, sizeof dst, "%s/%s", scratch_dir,
 				 mnt->path);
-		sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL);
+		// NOTE: we are not using sc_do_mount because we want non-fatal errors.
+		if (mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL) < 0) {
```

Comment:
I agree with @jdstrand. We shouldn't be duplicating this logic just to make it optional. The logic here is _already_ different compared to what exists in sc_mount, in multiple senses. That sounds like a recipe for awkward bugs.

I suggest having a sc_optional_do_mount function instead, and have both functions being wrappers to the same underlying logic, which may be a private function in libsnap-confine with the actual code.